### ISSUE TITLE
screenShare={true} not working

### ISSUE THREAD

**Issue by thegitparticle** [blackuy/react-native-twilio-video-webrtc#535]
### Steps to reproduce
1. Execute a simple video call code
2. Video and audio work fine, both the cameras work
3. But when putting screenShare={true}, nothing happens. 

### Expected behaviour
Screen recording must replace or co-exist along with the video call. 

### Actual behaviour
Nothing happens. Regular video call just happens and no change despite setting the boolean to true. I'd like to if I need to do any additional setup to the screen sharing work inside video calls

### Environment
- **Node.js version**: 14.15.1
- **React Native version**: 0.64.1
- **React Native platform + platform version**: iOS 14.5 and Android 10

### react-native-twilio-video-webrtc
**Version**: npm version or "master" latest, installed just today.

**Comment by agam-colaburate** [comment]
+1

**Comment by slycoder** [comment]
I think this feature was removed quite a while ago as part of a refactor and the docs weren't properly updated to reflect that it was removed.  

Anyone interested in re-adding this functionality?

**Comment by agam-colaburate** [comment]
@slycoder  Thanks for the clarification.
Do we have any alternative for screen sharing/streaming in react native ?

**Comment by slycoder** [comment]
Not sure, the easiest approach might to just add this functionality back in.  If you're up for it, we'd welcome the PR.

**Comment by italoj** [comment]
Hey @slycoder, do you have some documentation or something else that helps to implement the screen sharing again? Or do you have the PR that this functionality was removed?

**Comment by slycoder** [comment]
This functionality has never existed on android; I don't recall when we last had this working on iOS but it was a while ago.  If someone wants to contribute a fresh implementation, it would be welcome.  I don't imagine it would be too difficult but I don't have time at the moment to work on it.

**Comment by hamzaawais96** [comment]
Any update on it?

**Comment by hamzaawais96** [comment]
No worries. I have implemented "in-app screen capture" and screen broadcasting.

**Comment by vipulSoni458** [comment]
Hi hamzaawais96, can you please elaborate how you implemented screen capture. It would be really helpful

**Comment by vipulSoni458** [comment]
@hamzaawais96  Like I want to pass screenshare tracks from ios device to web. If you know anything which might help then it would really helpful

**Comment by hamzaawais96** [comment]
@vipulSoni458 I created a PR. You can take look into it.

**Comment by hamzaawais96** [comment]
@vipulSoni458 nope. This will share video tracks of local particpant and screen tracks both. The other remote particpant will be able to see local particpant video and screen sharing as well.

**Comment by vipulSoni458** [comment]
@hamzaawais96  Okay thanks. Will try to implemenent this. Thanks again. :)

**Comment by vipulSoni458** [comment]
Hi @hamzaawais96 I am able to screen share but suddenly it is start giving me this error , Exception 'The supplied delegate does not conform to the TVIAppScreenSourceDelegate protocol.' was thrown while invoking toggleScreenSharing on target TWVideoModule with params(1). Can you please me know how can I solve this error?
Thanks

**Comment by vipulSoni458** [comment]
@hamzaawais96  actually this issue is solved but when I turned off the Screen Share the my localParticipant video is also stopped and removed from videotracks, can you please suggest any idea on this how can I resolve this issue?
thanks

**Comment by hamzaawais96** [comment]
@vipulSoni458 it's working fine for me. On stop sharing, localParticipant video still work.

**Comment by Vipul4439** [comment]
Okay, I will check it again...its removing for me when component rerender

**Comment by vipulSoni458** [comment]
Hi @hamzaawais96 I am able to screen share but suddenly it is start giving me this error , Exception 'The supplied delegate does not conform to the TVIAppScreenSourceDelegate protocol.' was thrown while invoking toggleScreenSharing on target TWVideoModule with params(1). Can you please me know how can I solve this error?
Thanks

**Comment by dynamisch-manish** [comment]
https://www.appsloveworld.com/swift/100/211/screen-sharing-using-twilio-in-ios

@Vipul4439 @hamzaawais96

**Comment by mohdmunaf** [comment]
Screen sharing isn't possible on Android?

**Comment by msahu2595** [comment]
> No worries. I have implemented "in-app screen capture" and screen broadcasting in iOS.

Hi @hamzaawais96, with your PR I am only able to share my app's screen view.

I also want to share another app's screen.

Like mentioned in twilio's official documentation, that is possible using broadcast extension. 
https://www.twilio.com/docs/video/ios-v5-screen-capture#broadcast-extension-method 

Have you used that ??

if yes, could you create a PR or share your code?

**Comment by msahu2595** [comment]
> This functionality has never existed on android; I don't recall when we last had this working on iOS but it was a while ago. If someone wants to contribute a fresh implementation, it would be welcome. I don't imagine it would be too difficult but I don't have time at the moment to work on it.

Created pull request,
https://github.com/blackuy/react-native-twilio-video-webrtc/pull/663

**Comment by SaadAli-rolustech** [comment]
@hamzaawais96 the changes don't work as mentioned in the PR. i have tried exactly the way you have mentioned. still nothing happens. just shows the pop for permission and then nothing, no streaming to the other user, please advice
